tensorflow_datasets/core/object_visualizer.py

```diff
@@ -0,0 +1,178 @@
+      if isinstance(feature, features_lib.Image)
+  ]
+
+  if not image_keys:
+    return False
+
+  sequence_keys = [
+      k for k, feature in ds_info.features.items()
+      if isinstance(feature, features_lib.Sequence)
+  ]
+
+  bbox_keys = []
+  for key in sequence_keys:
+    if isinstance(ds_info.features[key].feature, features_lib.FeaturesDict):
+      bbox_keys = [
+          k for k, feature in ds_info.features[key].items()
+          if isinstance(feature, features_lib.BBoxFeature)
+      ]
+
+  if bbox_keys:
+    return True
+
+  return False
+
+def build(ds_info, ds, **kwargs):
+  """Builds the plot for visualization"""
+  rows = kwargs.setdefault('rows', 3)
+  cols = kwargs.setdefault('cols', 3)
+  plot_scale = kwargs.setdefault('plot_scale', 3)
+  image_key = kwargs.setdefault('image_key', None)
+  args = ['rows', 'cols', 'plot_scale', 'image_key']
+
+  for key in kwargs:
+    if key not in args:
+      raise TypeError(
+          "show_example() got an unexpected keyword argument '{}'".format(key)
+          )
+
+  plt = lazy_imports_lib.lazy_imports.matplotlib.pyplot
+
+  if not image_key:
+    # Infer the image and label keys
+    image_keys = [
+        k for k, feature in ds_info.features.items()
+        if isinstance(feature, features_lib.Image)
+    ]
+
+    if len(image_keys) > 1:
+      raise ValueError(
+          "Multiple image features detected in the dataset. Using the first one. You can "
+          "use `image_key` argument to override. Images detected: %s" %
+          (",".join(image_keys)))
+
+    image_key = image_keys[0]
+
+  # Infer the sequence keys since BBox are wrapped inside them
+  sequence_keys = [
+      k for k, feature in ds_info.features.items()
+      if isinstance(feature, features_lib.Sequence)
+  ]
+
+  # Infer the BBox keys in the sequence feature connectors
+  bbox_keys = []
+  for key in sequence_keys:
+    if isinstance(ds_info.features[key].feature, features_lib.FeaturesDict):
+      bbox_keys = [
+          k for k, feature in ds_info.features[key].items()
+          if isinstance(feature, features_lib.BBoxFeature)
+      ]
+      if len(bbox_keys) > 0:
+        sequence_key = key
+        break
+
+  # Infer the label keys in the sequence feature connector (BBox Labels)
+  label_keys = [
+      k for k, feature in ds_info.features[sequence_key].items()
+      if isinstance(feature, features_lib.ClassLabel)
+  ]
+
+  # Taking first label key since some datasets have multiple label keys
+  # for BBoxes as in voc dataset
+  label_key = label_keys[0] if len(label_keys) > 0 else None
+  if not label_key:
+    logging.info("Was not able to auto-infer label.")
+  bbox_key = bbox_keys[0] if len(bbox_keys) > 0 else None
+
+  num_examples = rows * cols
+  examples = list(dataset_utils.as_numpy(ds.take(num_examples)))
+
+  fig = plt.figure(figsize=(plot_scale*cols, plot_scale*rows))
+  fig.subplots_adjust(hspace=1/plot_scale, wspace=1/plot_scale)
+  for i, ex in enumerate(examples):
```

Comment:
Rather than duplicating all this between the visualizer, is there a way to reuse more the code ?